src/support.cpp

```diff
@@ -48,6 +48,302 @@ bool AreaSupport::handleSupportModifierMesh(SliceDataStorage& storage, const Set
+    for (unsigned int layer_nr = 0; layer_nr < total_layer_count - 1; ++layer_nr)
+    {
+        assert(storage.support.supportLayers[layer_nr].support_infill_parts.empty() && "support infill part list is supposed to be uninitialized");
+
+        Polygons& global_inner_support_areas = storage.support.supportLayers[layer_nr].inner_support_areas;
+        global_inner_support_areas.clear();
+
+        // this is the complete support areas on this layer
+        const Polygons& whole_support_areas = storage.support.supportLayers[layer_nr].supportAreas;
+
+        if (whole_support_areas.size() == 0 || layer_nr < min_layer || layer_nr > max_layer)
+        {
+            // initialize support_infill_parts empty
+            storage.support.supportLayers[layer_nr].support_infill_parts.clear();
+            continue;
+        }
+
+        std::vector<PolygonsPart> support_islands = whole_support_areas.splitIntoParts();
+        for (unsigned int island_idx = 0; island_idx < support_islands.size(); ++island_idx)
+        {
+            storage.support.supportLayers[layer_nr].support_infill_parts.emplace_back();
+            SupportInfillPart& support_infill_part = storage.support.supportLayers[layer_nr].support_infill_parts.back();
+            support_infill_part.infill_areas_per_combine_per_density.clear();
+            support_infill_part.outline = support_islands[island_idx];
+            support_infill_part.insets.clear();
+            support_infill_part.inner_infill_areas.clear();
+
+            support_infill_part.infill_overlap = 0;
+            if (support_pattern == EFillMethod::GRID || support_pattern == EFillMethod::TRIANGLES || support_pattern == EFillMethod::CONCENTRIC)
+            {
+                support_infill_part.infill_overlap = infill_overlap; // support lines area should be expanded outward to overlap with the boundary polygon
+            }
+
+            assert(support_infill_part.infill_areas_per_combine_per_density.size() == 0);
+            if (wall_line_count > 0)
+            {
+                // generate insets and use the first inset as the infill area
+                AreaSupport::generateOutlineInsets(
+                    support_infill_part.insets,
+                    support_infill_part.outline,
+                    wall_line_count,
+                    support_line_width);
+                if (support_infill_part.insets.empty())
+                {
+                    continue;
+                }
+                support_infill_part.inner_infill_areas = support_infill_part.insets[0];
+            }
+            else
+            {
+                // for zig zag, we don't want a wall, so we use the outline as the inner infill areas
+                support_infill_part.inner_infill_areas = support_infill_part.outline;
+            }
+
+            // also generate the global inner_infill_area
+            global_inner_support_areas.add(support_infill_part.inner_infill_areas);
```

Comment:
Why are you collecting this here?
Please remove `SupportLayer::inner_infil_areas`.
It will needlessly consume memory.